source/performance/F0.Minesweeper.Logic.Benchmarks/Program.cs

```diff
@@ -0,0 +1,20 @@
+using System.Diagnostics;
+using BenchmarkDotNet.Columns;
+using BenchmarkDotNet.Configs;
+using BenchmarkDotNet.Diagnosers;
+using BenchmarkDotNet.Running;
+
+IConfig config = Debugger.IsAttached
+	? CreateDebugConfiguration()
+	: CreateBenchmarkConfiguration();
+
+_ = BenchmarkSwitcher.FromAssembly(typeof(Program).Assembly).Run(args, config);
+
+static IConfig CreateBenchmarkConfiguration()
+	=> DefaultConfig.Instance
+		.WithOptions(ConfigOptions.StopOnFirstError)
+		.AddColumn(StatisticColumn.Min, StatisticColumn.Max, StatisticColumn.Median, RankColumn.Arabic)
+		.AddDiagnoser(MemoryDiagnoser.Default);
```

Comment:
Out of curiosity: would there be any way where I could still opt-out of a globally set configuration if I want to run a benchmark where for example I'm 100% sure the memory is not of any concern?

Reply:
Not that I know of. We could remove the `MemoryDiagnoser` from the default config, and than add it to benchmark classes explicitly via the attribute `[MemoryDiagnoser]` where it makes sense.

Reply:
We keep it for now as default, if we see a `[Benchmark]`s which actually do not require it, we may change that later.
On the other hand, as a "safety" mechanism, it may always show potentially unwanted allocations.